### PR TITLE
TST Ensure that sklearn/cluster/tests/test_k_means.py is seed insensitive

### PR DESCRIPTION
**Pull request by ShoaibKhan**
#### Reference Issues/PRs

Fixes sklearn/sklearn/cluster/tests/test_k_means.py in https://github.com/scikit-learn/scikit-learn/issues/22827 upstream.

#### What does this implement/fix? Explain your changes.

This upgrades the k_means tests to account for greater randomness using global_random_seed.

#### Any other comments?

Some tests were left unchanged, as suggested in [the original issue](https://github.com/scikit-learn/scikit-learn/issues/22827), as they don't test mathematical properties. Many tests failed with the introduction of randomness and to fix this, they were left as is.

### CODE REVIEW

**Review by tahmid-haque — CHANGES_REQUESTED** [review]
Not sure if my branch is checkout out wrong but when running the tests a bunch failed. I ran them all at once using 
```
SKLEARN_TESTS_GLOBAL_RANDOM_SEED="all" pytest sklearn/cluster/tests/test_k_means.py -k 'test_kmeans_convergence or test_minibatch_update_consistency or test_all_init or test_minibatch_kmeans_partial_fit_init or test_fortran_aligned_data or test_minibatch_kmeans_verbose or test_kmeans_verbose or test_minibatch_sensible_reassign or test_minibatch_reassign or test_minibatch_with_many_reassignments or test_minibatch_declared_convergence or test_minibatch_iter_steps or test_kmeans_copyx or test_score_max_iter or test_predict or test_dense_sparse or test_predict_dense_sparse or test_fit_transform or test_n_init or test_float_precision or test_weighted_vs_repeated or test_scaled_weights or test_result_equal_in_diff_n_threads or test_k_means_1_iteration or test_euclidean_distance or test_inertia or test_sample_weight_unchanged or test_kmeans_plusplus_dataorder'
``` 
The command took 15 minutes to run :( Can you double check that they all pass? If not, I'd consider whether certain tests are worth converting here.

**Comment by ShoaibKhan** [comment]
> Not sure if my branch is checkout out wrong but when running the tests a bunch failed. I ran them all at once using

Right you are, I didn't account for some tests. 

> Can you double check that they all pass? If not, I'd consider whether certain tests are worth converting here.

I'll double check all these pass and update accordingly

**Review by tahmid-haque — APPROVED** [review]
All good!

**Review by Tony-beeper — APPROVED** [review]
LGTM!

**Comment by ogrisel** [comment]
Hello @ShoaibKhan, did you mean to do a PR to the scikit-learn main repo or to @TheBicPen's fork?

If you expect this PR to be merged to scikit-learn, you should open a new pull-request targeting the `main` branch of the main repo (https://github.com/scikit-learn/scikit-learn) instead of this fork.

**Comment by ShoaibKhan** [comment]
> Hello @ShoaibKhan, did you mean to do a PR to the scikit-learn main repo or to @TheBicPen's fork?
> 
> If you expect this PR to be merged to scikit-learn, you should open a new pull-request targeting the `main` branch of the main repo (https://github.com/scikit-learn/scikit-learn) instead of this fork.

Hi @ogrisel, this PR was just meant for @TheBicPen's fork, but I have made another PR on the main fork.
